### PR TITLE
Add an additional OSX_SDK_DIR for newer macOS images on Azure

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Add an additional `OSX_SDK_DIR` for newer macOS images on Azure.
ref: https://github.com/conda-forge/python-feedstock/pull/402#issuecomment-708351485

### CODE REVIEW

File: recipe/download_osx_sdk.sh [hunk 1]
```diff
@@ -26,7 +26,10 @@ if [[ "$MACOSX_SDK_VERSION" == "11.0" ]]; then
     if [[ "$CI" == "travis" ]]; then
         export OSX_SDK_DIR=/Applications/Xcode-12.for.macOS.Universal.Apps.beta.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
     elif [[ "$CI" == "azure" ]]; then
-        export OSX_SDK_DIR=/Applications/Xcode_12_beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
+        export OSX_SDK_DIR=/Applications/Xcode_12.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
+        if [[ ! -d "${OSX_SDK_DIR}" ]]; then
+            export OSX_SDK_DIR=/Applications/Xcode_12_beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
```
Comment (mbargull) [inline recipe/download_osx_sdk.sh:31]: I retained this one just in case an older image could be used.
This can be removed in the future.

**Review by chrisburr — APPROVED** [review]
I was just about to do this as it's affecting https://github.com/conda-forge/clangdev-feedstock/pull/106 as well

**Comment by beckermr** [comment]
Let's wait for @isuruf but lgtm!

**Comment by isuruf** [comment]
Thanks

**Pull request merged**